src/calculations.cpp

```diff
@@ -76,4 +78,41 @@ double distSquared(const tf2::Transform & a, const tf2::Transform & b)
   return a.getOrigin().distance2(b.getOrigin());
 }
 
+tf2::Transform closestPoseOnSegment(
+  const tf2::Transform & point, const tf2::Transform & segment_start,
+  const tf2::Transform & segment_end, bool estimate_pose_angle)
+{
+  const double l2 = distSquared(segment_start, segment_end);
+  if (l2 == 0) {
```

Comment:
Is it safe to compare a double to an exact value?

Reply:
In this case, I think it is.

We need `l2` not to be equal to 0 so we can divide by it to calculate `t`. If `l2` is nearly 0 but not quite, `t` would be very large but it is clamped between 0 and 1. So, as far as I can tell, the check should not be an issue.

Reply:
Also, this is not a change. The implementation is the same as it was before this PR, just moved.